test/tests/constraints/equal_value_embeded_constraint/1D_2D_kinematic.i

```diff
@@ -0,0 +1,81 @@
+###########################################################
+# This is a test that demonstrates a user-defined
+# constraint. It forces variables in overlapping portion of
+# two blocks to have the same value
+###########################################################
+
+
+[Mesh]
+  file = gold/1D_2D.e
+[]
+
+[Variables]
+  [phi]
+    order = FIRST
+    family = LAGRANGE
+  []
+[]
+
+[Kernels]
+  [diffusion]
+    type = Diffusion
+    variable = phi
+  []
+[]
+
+
+[BCs]
+  [top]
+    type = DirichletBC
+    variable = phi
+    boundary = 1
+    value = 10.0
+  []
+  [bottom]
+    type = DirichletBC
+    variable = phi
+    boundary = 2
+    value = 0.0
+  []
+  [left]
+    type = DirichletBC
+    variable = phi
+    boundary = 3
+    value = 10.0
+  []
+  [right]
+    type = DirichletBC
+    variable = phi
+    boundary = 4
+    value = 0.0
+  []
+[]
+
+[Constraints]
+  [equal]
+    type = EqualValueEmbededConstraint
+    slave = 2
+    master = 1
+    penalty = 1e3
+    master_variable = phi
+    variable = phi
+    #formulation = penalty
+  []
+[]
+
+[Executioner]
+  type = Steady
+  solve_type = PJFNK
+  petsc_options_iname = '-pc_type -ksp_gmres_restart'
+  petsc_options_value = 'lu       101'
```

Comment:
This set of solver options won't work in parallel. Try just removing `petsc_options_iname` and `petsc_options_value` to use the default options, which will probably work fine here.

Reply:
when I use the default options, tests using formulation = penalty do not converge.

Reply:
If I just use -pc_type = lu it converges.